drivers/python/rethinkdb/net_twisted.py

```diff
@@ -156,67 +141,40 @@ def clientConnectionFailed(self, connector, reason):
         pass
 
 class CursorItems(DeferredQueue):
-
     def __init__(self):
-        super(CursorItems, self).__init__(self)
-
-    @inlineCallbacks
-    def wait_for_new_item(self):
-        """
-        Callback whenever a new item arrived.
-        It use the behavior of put.
-        Warning: It ignores QueueUnderflow / QueueOverflow mechanisms.
-        """
-        d = Deferred(canceller=self._cancelGet)
-        self.waiting.append(d)
-        item = yield d
-        self.pending.append(item)
-        returnValue(item)
+        super(CursorItems, self).__init__()
 
     def cancel_getters(self, err):
         """
         Cancel all waiters.
         """
         for waiter in self.waiting[:]:
-            waiter.errback(err)
+            if not waiter.called:
+                waiter.errback(err)
             self.waiting.remove(waiter)
 
+    def extend(self, data):
+        for k in data:
+            self.put(k)
+
     def __len__(self):
         return len(self.pending)
 
-    def __getitem__(self, key):
-        return self.pending[key]
-
 class TwistedCursor(Cursor):
-
     def __init__(self, *args, **kwargs):
         super(TwistedCursor, self).__init__(*args, **kwargs)
         self.items = CursorItems()
-        self.new_response = Deferred()
-
-    def _extend_queue(self, data):
-        for k in data:
-            self.items.put(k)
+        self.waiting = list()
 
     def _extend(self, res):
```

Comment:
This function was duplicating the behavior of the base class `_extend` - this now calls `Cursor._extend`, then performs any specialized behavior (canceling/calling Deferreds).